dataprofiler/tests/profilers/test_unstructured_text_profile.py

```diff
@@ -164,4 +165,95 @@ def test_case_sensitivity(self):
             expected_word_count = {'hello': 1, 'name': 1, 'grant': 2, 'bob': 1,
                                    'friends': 1}
             self.assertDictEqual(expected_word_count, profile['word_count'])
-        
\ No newline at end of file
+
+    def test_text_profile_with_wrong_options(self):
+        with self.assertRaisesRegex(ValueError,
+                "TextProfiler parameter 'options' must be of type"
+                " TextProfilerOptions."):
+            TextProfiler("Name", options="wrong_data_type")
+
+    def test_options_default(self):
+        options = TextProfilerOptions()
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'sentence': 1, 'Test': 1, 'test': 1}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+    def test_options_case_sensitive(self):
+        options = TextProfilerOptions()
+        options.is_case_sensitive = False
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'sentence': 1, 'test': 2}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+    def test_options_stop_words(self):
+        # with a list of stopwords
+        options = TextProfilerOptions()
+        options.stop_words = ['hello', 'sentence', 'is', 'a']
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'This': 1, 'Test': 1, 'test': 1}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+        # with an empty list
+        options = TextProfilerOptions()
+        options.stop_words = []
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'This': 1, 'is': 1, 'test': 1}
+        self.assertDictEqual(expected_word_count, profile['word_count'])
```

Comment:
good, next time though, can just stick with the same sentence so we have consistency and all that changes is the stop words

Reply:
changed the sample to be consistent with other tests. The reason I changed it previously is to make the sample simple enough to read.